File: frontend/src/types/customer.ts
import type { Priority } from "./common"

export type Segment =
  | "Student"
  | "Young Professional"
  | "Family"
  | "Business"
  | "Retired"
  | "High Net Worth"

export type Country = "India" | "USA" | "UK" | "Singapore" | "Australia"
export type CustomerStatus = "Active" | "Inactive" | "Dormant"
export type KycStatus = "Verified" | "Pending" | "Rejected"

export interface BehaviourEvent {
  id: string
  date: string
  type: string
  description: string
  channel: string
}

export interface TransactionEvent {
  id: string
  date: string
  description: string
  category: string
  amount: number
  direction: "credit" | "debit"
  merchant: string
}

export interface JourneyStage {
  stage: string
  date: string
  status: "completed" | "current" | "upcoming"
  description: string
}

export interface CustomerRecommendation {
  id: string
  product: string
  priority: Priority
  reason: string
  triggeredAgent: string
  status: "Pending" | "Accepted" | "Rejected"
  generatedAt: string
}

export interface CustomerSummary {
  id: string
  name: string
  email: string
  phone: string
  avatarSeed: string
  segment: Segment
  country: Country
  status: CustomerStatus
  kycStatus: KycStatus
  accountBalance: number
  riskScore: number
  financialHealthScore: number
  joinDate: string
}

export interface CustomerDetail extends CustomerSummary {
  behaviourTimeline: BehaviourEvent[]
  transactionTimeline: TransactionEvent[]
  aiRecommendations: CustomerRecommendation[]
  journey: JourneyStage[]
}

export interface CustomerListResponse {
  items: CustomerSummary[]
  total: number
  page: number
  pageSize: number
}

export interface CustomerListParams {
  search?: string
  segment?: string
  country?: string
  status?: string
  kycStatus?: string
  page?: number
  pageSize?: number
}

File: backend/app/schemas/customer.py
from typing import Literal, Optional
from pydantic import BaseModel

Segment = Literal[
    "Student", "Young Professional", "Family", "Business", "Retired", "High Net Worth"
]
Country = Literal["India", "USA", "UK", "Singapore", "Australia"]
CustomerStatus = Literal["Active", "Inactive", "Dormant"]
KycStatus = Literal["Verified", "Pending", "Rejected"]


class BehaviourEvent(BaseModel):
    id: str
    date: str
    type: str
    description: str
    channel: str


class TransactionEvent(BaseModel):
    id: str
    date: str
    description: str
    category: str
    amount: float
    direction: Literal["credit", "debit"]
    merchant: str


class JourneyStage(BaseModel):
    stage: str
    date: str
    status: Literal["completed", "current", "upcoming"]
    description: str


class CustomerRecommendation(BaseModel):
    id: str
    product: str
    priority: Literal["High", "Medium", "Low"]
    reason: str
    triggeredAgent: str
    status: Literal["Pending", "Accepted", "Rejected"]
    generatedAt: str


class CustomerSummary(BaseModel):
    id: str
    name: str
    email: str
    phone: str
    avatarSeed: str
    segment: Segment
    country: Country
    status: CustomerStatus
    kycStatus: KycStatus
    accountBalance: float
    riskScore: int
    financialHealthScore: int
    joinDate: str


class CustomerDetail(CustomerSummary):
    behaviourTimeline: list[BehaviourEvent]
    transactionTimeline: list[TransactionEvent]
    aiRecommendations: list[CustomerRecommendation]
    journey: list[JourneyStage]


class CustomerListResponse(BaseModel):
    items: list[CustomerSummary]
    total: int
    page: int
    pageSize: int
